PulseSampleApp/app/src/main/java/com/ooyala/sample/players/PulsePlayerActivity.java

```diff
@@ -120,4 +128,18 @@ public VideoItem getVideoItem() {
 		videoItem.setContentCode(getIntent().getExtras().getString("embedCode"));
 		return videoItem;
 	}
+
+	private void destroyPlayer() {
+		if (player != null) {
+			player.destroy();
+			player = null;
+		}
+		if (skinLayout != null) {
+			skinLayout.release();
+		}
+		if (playerSkinLayoutController != null) {
+			playerSkinLayoutController.deleteObserver(this);
```

Comment:
Why we don't call onDestroy() method of playerSkinLayoutController as early?

Reply:
done